### PR TITLE
Add guidance layer examples

### LINKED ISSUE #1926: Button Guidance - Add Guidance and update examples for how to close a Layer
- [x] Update all Layer examples to align with new guidance around how to close a layer
- [x] Add/Adjust written guidance on site

Guidance + example changes can be found [here](https://www.figma.com/file/Ts53TAipMolmsv9DxWG3p0/HPE-Layer-Component?node-id=2%3A14).
This ticket stems from #1761

### PR DESCRIPTION
**Pull request by britt6612**
Figma link [here](https://www.figma.com/file/Ts53TAipMolmsv9DxWG3p0/HPE-Layer-Component?node-id=2%3A14)
#### What does this PR do?
Adds the new changes for layer 
#### Where should the reviewer start?
layer.js
#### What testing has been done on this PR?
manually layer page
#### Any background context you want to provide?
Reverted Prism back to 1.24.0 
I also created an issue on their backlog may help us determine why when upgrading to 1.25.0 it broke
#### What are the relevant issues?
closes #1926 
#### Should this PR be mentioned in Design System updates?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: aries-site/src/examples/cardPreviews/code-blocks.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ import React from 'react';
 
 import { Box } from 'grommet';
 import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
-import { useDarkMode } from '../../utils';
 import { prism } from 'grommet-theme-hpe';
+import { useDarkMode } from '../../utils';
```
Comment (halocline) [inline aries-site/src/examples/cardPreviews/code-blocks.js:6]: 👍

File: aries-site/src/examples/components/layer/LayerSideDrawerExample.js [hunk 2]
```diff
@@ -131,8 +130,6 @@ export const LayerSideDrawerExample = () => {
           position="right"
           full={size !== 'small' ? 'vertical' : true}
           modal={false}
```
Comment (halocline) [inline aries-site/src/examples/components/layer/LayerSideDrawerExample.js:132]: The Figma file shows this as being a modal.

On a tangent, if we make this a modal as well, then all examples are modal layers. Do we have use cases where the layer should not be a modal? Should we be showing them? Do we have guidance around when the page content should remain active vs. when a modal should be used to disable interaction?

Comment (halocline) [reply aries-site/src/examples/components/layer/LayerSideDrawerExample.js:132]: It looks like we have the guidance above the example, but the example is not following the guidance completely.

File: aries-site/src/pages/components/layer.mdx [hunk 3]
```diff
@@ -52,6 +52,28 @@ For example, implementation of this "Close" button for a Layer would be:
...
+- For user generated layers such as directly opened dialog moodel, the Layer can be closed by clicking outside of it's coontent. 
+- For system generated layers such as Toast Notification, the layer cannot be closed by clicking outside of it's content.
+
+### Actionable
+Provides users with any type of action, required or optional. These can be confirmation, filters, forms, etc.
+
+Actionable Layers contain the following elements and interactions: 
+- A cancel or close button will always be positioned within the footer of the Layer. Note that the "X" icon is omitted to avoid any confusion the user may have on which option they should use to close out the Layer. 
+- The buttons and the Layer heading will live above all the other content and be sticked during scrolling. 
+- The Layer cannot be closed by clicking outside of it's content. This is to prevent accidental misclicks.
```
[the remark this answers is not in the document]
Comment (halocline) [reply aries-site/src/pages/components/layer.mdx:73]: ☝️  - yep. I was getting a bit rhetorical...

File: aries-site/package.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
     "grommet-icons": "https://github.com/grommet/grommet-icons/tarball/stable",
     "grommet-theme-hpe": "https://github.com/grommet/grommet-theme-hpe/tarball/stable",
     "next-compose-plugins": "^2.2.0",
-    "prismjs": "^1.19.0",
+    "prismjs": "1.24.0",
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/package.json:16]: Yes I can!

File: aries-site/src/pages/components/layer.mdx [hunk 5]
```diff
@@ -52,6 +52,28 @@ For example, implementation of this "Close" button for a Layer would be:
...
 
+## Types 
+
+All Layers can be categorized into two different types based on the content being displayed, informational or Actionable. This is important to note, as this affects how the user can close the Layer.
+
+### Informational
+Used purely to inform users of status or state. These are text-only with no required or optional actions. These can be dialog modals, toast notificatioons, etc.
+
+Informational Layers contain the following elements and interactions:
+- A sole "X" icon for the user to close out of the Layer. All oother elements such as a close buttoon woould onnly cause further clutter and confusion for the user.
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/pages/components/layer.mdx:63]: Ill look back I think the figma needs to be updated as well.

File: aries-site/src/pages/components/layer.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/pages/components/layer.mdx:73]: @KennyAtHPE was going to update this porition

File: aries-site/src/pages/components/layer.mdx [hunk 6]
```diff
@@ -52,6 +52,28 @@ For example, implementation of this "Close" button for a Layer would be:
...
+## Types 
+
+Depending on the content displayed, Layers may be thought of as either [informational](components/layer#informational) or [actionable](components/layer#actionable). This distinction is important as it affects whether the page content adjacent to the Layer should be interactive, as well as impacting how a user may close the Layer.
+
+### Informational
+Used purely to inform users of status or state. These are text-only with no required or optional actions. These may be [dialog modals](/components/layer#center-modal), [toast notifications](/templates/toast-notifications), etc.
+
+Informational Layers contain the following elements and interactions:
+- A sole ["X" icon](https://icons.grommet.io/?s=formclose) for the user to close out of the Layer. Additional elements, such as a close button, cause further clutter and confusion for the user.
+- For user generated layers such as directly opened dialog model, the Layer can be closed by clicking outside of it's content.
```
Comment (taysea) [inline aries-site/src/pages/components/layer.mdx:64]:
```suggestion
- For dialog modals that open as a result of a user clicking an element on the page, the Layer can be closed by clicking outside of its content. 
```
I'm not fully clear on what this means -- suggesting a wording change based on how i'm interpreting it.

File: aries-site/src/pages/components/layer.mdx [hunk 7]
```diff
@@ -28,9 +28,9 @@ Layer is used to define the properties and behaviors of an overlay for things li
 
 - Modal is best applied when you need to center attention on specific content, a task, or behavior and not allow interactivity with the underlying content.
 - Position forms to the right (side-drawer).
-- Position notifications top or bottom.
-- Use a close control to guide users to move out of the layer. Provide an alternative to move out of the layer by allowing use of the 'esc' key or clicking off the container of the layer.
+- Positioning of notification layers depends on the notification's context. Please refer to [notification guidance](/components/notification) for more detail.
 - Layer is not great for navigation as it 'hides' content. Maintain context for the user with layer.
+- Provide an alternative to move out of the layer by allowing use of the 'esc' key. This applies to all Layers, regardless of type (e.g. [informational](/components/layer#informational) or [actionable](/components/layer#actionable)).
```
Comment (taysea) [inline aries-site/src/pages/components/layer.mdx:33]:
```suggestion
- Provide an alternative to move out of the Layer by allowing use of the 'esc' key. This applies to all Layers, regardless of type (e.g. [informational](/components/layer#informational) or [actionable](/components/layer#actionable)).
```

**Pull request merged**